migrations/custom_migration_factory.go

```diff
@@ -0,0 +1,61 @@
+package migrations
+
+import (
+	"github.com/mongodb/anser"
+	"github.com/mongodb/anser/model"
+	"github.com/pkg/errors"
+)
+
+const perfCopyVariantMigrationFactory = "perf-copy-variant-factory"
+
+type migrationGeneratorFactoryFactory func(args map[string]string) migrationGeneratorFactory
+
+var CustomMigrationRegistry = map[string]migrationGeneratorFactoryFactory{
+	perfCopyVariantMigrationFactory: perfCopyVariantFactoryFactory,
+}
+
+// NewCustomApplication constructs and sets up an application instance from
+// a configuration structure and a migration factory.
+func NewCustomApplication(env anser.Environment, conf *CustomConfiguration, factory map[string]migrationGeneratorFactoryFactory) (*anser.Application, error) {
```

Comment:
On the contrary: A `migrationGeneratorFactoryFactory` returns a `migrationGeneratorFactory`, as expected. The function is useful, because it allows lookups in the `CustomMigrationRegistry` (https://github.com/evergreen-ci/evergreen/pull/1141/files/a901ad6a724ced82594a76dfe83fca8e1e623ca7#diff-24edbd75fbd8471c4751d9ac12973b13R33).  Users specify the custom migration they would like to run in the configuration file they pass.